### PR TITLE
Upstreaming WASM friendliness and other fixes to 80bit branch

### PR DESCRIPTION
**Pull request by RiverRuby**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -20,29 +20,32 @@ pub use backend::{
...
     }
 
     fn get_row_slice(&self, row_idx: usize) -> &[Self::MatElement] {
         self.as_ref()[row_idx].as_ref()
     }
 
-    fn iter_rows(&self) -> impl Iterator<Item = &Self::R> {
-        self.as_ref().iter().map(move |r| r)
+    fn iter_rows(&self) -> RowIterator<'_, Self::R> {
+        Box::new(self.as_ref().iter().map(move |r| r))
```
Comment (ChihChengLiang) [inline src/lib.rs:48]: Can you share reasons on why adding boxes in get_row, iter_rows, and iter_rows_mut?

**Review by ChihChengLiang — APPROVED** [review]
LGTM!
The hiring example looks really cool for a FHE demo.
I added a question, but it does not block merging.

File: src/lib.rs (hunk 1 above)
Comment (RiverRuby) [reply src/lib.rs:48]: oh yeah, for some reason i'm not able to build without doing this. I'm getting some error like a function returning `impl Iterator` isn't available in my version of Rust?

Comment (RiverRuby) [reply src/lib.rs:48]: are you guys using nightly rust? or what version? @ChihChengLiang

**Pull request merged**